### ISSUE TITLE
OH3.0.0: Ephemeris losing entered data on restart

### ISSUE THREAD

**Issue by DrTron** [openhab/openhab-core#2024]
I only noticed this as I have a rule requiring ephemeris:

On OH3 (3.0.0 - Release Build), setting ephemeris data in MainUI (country-region) takes several entries to "stick".
After that, ephemeris works and no errors show in the logs. However, after a restart of OH3, the fields are empty again and ephemeris does not work again until those fields are re-entered.

**Comment by DrTron** [comment]
Happened again, this time after adding a new zwave device, no OH3 restart. Just FYI.

**Comment by DrTron** [comment]
Still happening in OH 3.1.0 M2.

**Comment by curlyel** [comment]
Same here :-(

I've noticed, that after multiple restart attempts, the code for the entered region got multiple quotes in the UI, e.g.:

Instead of `Hesse` i get `""""""""he""""""""`

It seems to me, that the content of `/var/lib/openhab/config/org/openhab/ephemeris.config` is messed up every time of restart and another pair of quotes were added. This might prevent the content from been parsed correctly which leads to a non-working ephemeris feature.

Til we get a fix for this, I decided to add a config file in the "old" location which seems to work for me:
```
/etc/openhab/services# more ephemeris.cfg
country=de
region=he
dayset-school=[MONDAY,TUESDAY,WEDNESDAY,THURSDAY,FRIDAY]
dayset-weekend=[SATURDAY,SUNDAY]
```

**Comment by DrTron** [comment]
Thanks for the heads-up. I also notice the "accumulating quotes" in the UI.
I'm rather surprised that OH3 will accept a config file in the old location, but I'll certainly give it a try.

Are you sure "dayset-school" is correct? I was expecting it to be "dayset-workday". But maybe they work the same way.

**Comment by curlyel** [comment]
> "dayset-workday"

Mmh. Good catch ...
Honestly, I've copied over the content of var/lib/openhab/config/org/openhab/ephemeris.config and adjusted it to the old file syntax. There I had "dayset-school" for whatever reason...

But you are right, in the documentation is `dayset-workday=...`. Since I mainly use ephemeris for Rich's `timeOfDay` rule, It should not make any difference for me as I just have to distinguish between "weekend" and "default"

**Comment by DrTron** [comment]
Yes, probably not a difference as work/schoolday are treated as default.
I've also been using the "TimeOfDay" rule, can't really think of anyone using ephemeris for anything different ;-)

Best wishes to my old home state, Hesse!

**Comment by rkoshak** [comment]
Just a note. That config file is where you can define your own custom daysets. So dayset-school is fine. It's in the config file to show how to define a custom dayset. You can use that for something like using Ephemeris to track trash days and the like. Weekend is the only dayset that is defined by default through the UI. But it is not the only possible dayset that can be defined. 

Also, daysets can overlap. So you could have weekend defined as SAT and SUN and dayset-workday set to WED, THR, FRI, SAT and it'd be fine. 

As for the config file in the old location, text based config in OH 3 remains largely unchanged from what it was in OH 2.5. And except for MainUI stuff, there is a text based config file one can define to configure everything.

As for uses of Ephemeris, you can define conditions in UI Rules to only allow a rule to run on certain types of days (e.g. don't run this rule on Holidays, only run this rule on Weekends, etc.). In a rule you can do something different based on the type of day it is.

**Comment by DrTron** [comment]
I suppose the greatest benefit of a config file here is to define country and region to prevent it getting messed up all the time. I've not found the time yet, but I'll try that and see if it improves things.

**Comment by DrTron** [comment]
Just as a heads-up, using a config file worked out for me, too.
